File: CustomerStorage.ViewModels/CustomerModels/CreateCustomerViewModel.cs
using System.ComponentModel.DataAnnotations;

namespace CustomerStorage.ViewModels.CustomerModels
{
    public class CreateCustomerViewModel
    {
        [Required(ErrorMessage = "Name - is required")]
        public string Name { get; set; }

        [Required(ErrorMessage = "Company name - is required")]
        public string? CompanyName { get; set; }

        [Phone(ErrorMessage = "Invalid Phone")]
        [StringLength(13, ErrorMessage = "Invalid Phone")]
        public string? Phone { get; set; }

        [Required(ErrorMessage = "Email - is required")]
        [EmailAddress(ErrorMessage = "Invalid Email")]
        public string? Email { get; set; }
    }
}

File: CustomersStorage/client/customer_storage_angular/src/app/models/customer/create-customer-view-model.ts
export class CreateCustomerViewModel {
    name: string;
    companyName: string;
    phone: string;
    email: string;

    constructor(name: string, companyName: string, phone: string, email: string) {
        this.name = name,
        this.companyName = companyName,
        this.phone = phone,
        this.email = email
    }
}